File: 48. Rotate Image.cpp
#include<bits/stdc++.h>
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n=matrix.size();
        // int =matrix[0].size()-1;
        
    for(int i=0; i<n; i++){
     for(int j=0; j<n-i; j++){
         swap(matrix[i][j], matrix[n-1-j][n-1-i]);
     }
    }

   //Rotate the matrix about middle row
   for(int i=0; i<n/2; i++){
     for(int j=0; j<n; j++){
        swap(matrix[i][j], matrix[n-1-i][j]);
     }
   }
    }
};

File: 48. Rotate Image.java
class Solution {
    public void rotate(int[][] matrix) {
        int rowLen = matrix.length;
        int colLen = matrix[0].length;

        for(int i = 0; i < rowLen ; i++){
            for(int j = i + 1; j < colLen; j++){
                int temp = matrix[i][j] ;
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp ;
            }
        }

        
        for(int i = 0; i < rowLen ; i++){
            for(int j = 0; j < colLen/2; j++){
                int temp = matrix[i][j] ;
                matrix[i][j] = matrix[i][colLen-j-1];
                matrix[i][colLen-j-1] = temp ;
            }
        }

    }
}
